psse/psse-model/src/main/java/com/powsybl/psse/model/PsseRawReader.java

```diff
@@ -85,122 +321,108 @@ public void handleError(DataProcessingException error, Object[] inputRow, Parsin
-        int i = 0;
-        while (i < records.size()) {
-            PsseTransformer transformer = new PsseTransformer();
-            transformer.setFirstRecord(parseRecord(records.get(i++), PsseTransformer.FirstRecord.class));
-            transformer.setSecondRecord(parseRecord(records.get(i++), PsseTransformer.SecondRecord.class));
-            transformer.setThirdRecord1(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
-            transformer.setThirdRecord2(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
-            if (transformer.getFirstRecord().getK() != 0) {
-                transformer.setThirdRecord3(parseRecord(records.get(i++), PsseTransformer.ThirdRecord.class));
+    private static String readLineAndRemoveComment(BufferedReader reader) throws IOException {
+        String line = reader.readLine();
+        if (line == null) {
+            return null;
+        }
+        StringBuffer newLine = new StringBuffer();
+        Pattern p = Pattern.compile("('[^']+')|( )+");
+        Matcher m = p.matcher(removeComment(line));
+        while (m.find()) {
+            if (m.group().contains("'")) {
+                m.appendReplacement(newLine, m.group());
+            } else {
+                m.appendReplacement(newLine, " ");
             }
-            transformers.add(transformer);
         }
-
-        return transformers;
+        m.appendTail(newLine);
+        return newLine.toString().trim();
     }
 
-    public PsseRawModel read(BufferedReader reader) throws IOException {
-        Objects.requireNonNull(reader);
-
-        // case identification
-        PsseCaseIdentification caseIdentification = readCaseIdentification(reader);
-
-        PsseRawModel model = new PsseRawModel(caseIdentification);
-
-        // bus data
-        model.getBuses().addAll(parseRecords(readRecordBlock(reader), PsseBus.class));
-
-        // load data
-        model.getLoads().addAll(parseRecords(readRecordBlock(reader), PsseLoad.class));
-
-        // fixed shunt data
-        model.getFixedShunts().addAll(parseRecords(readRecordBlock(reader), PsseFixedShunt.class));
-
-        // generator data
-        model.getGenerators().addAll(parseRecords(readRecordBlock(reader), PsseGenerator.class));
-
-        // non transformer data
-        model.getNonTransformerBranches().addAll(parseRecords(readRecordBlock(reader), PsseNonTransformerBranch.class));
-
-        // transformer data
-        model.getTransformers().addAll(readTransformers(reader));
-
-        // area data
-        model.getAreas().addAll(parseRecords(readRecordBlock(reader), PsseArea.class));
+    // Read fields
 
-        // 2-terminal DC data
-        readRecordBlock(reader); // TODO
-
-        // voltage source converter data
-        readRecordBlock(reader); // TODO
+    private static String[] readFields(List<String> records, String[] headers, String delimiter) {
+        if (records.isEmpty()) {
+            return new String[] {};
```

Comment:
Since you use ArrayUtils, why don't use ArrayUtils.EMPTY_STRING_ARRAY?

Reply:
Done